src/Ssh/Ssh/Common/SshBaseCmdlet.cs

```diff
@@ -626,6 +669,160 @@ protected internal bool IsArc()
 
         #region Private Methods
 
+        #region Get Relay Information Private Methods
+
+        /// <summary>
+        /// Checks if the target port matches the service configuration port 
+        /// </summary>
+        /// <returns> False if target port isn't allowed in the service configuration, True if it is or if we don't have permission to determine that.</returns>
+        private bool ServiceConfigurationMatchesTargetPort()
+        {
+            ServiceConfigurationResource result;
+            try
+            {
+                result = ServiceConfigurationsClient.GetAsync(ResourceId, "default", "SSH").GetAwaiter().GetResult();
+            }
+            catch (PowerShell.Ssh.Helpers.HybridConnectivity.Models.ErrorResponseException)
+            {
+                // This will more often than not happen when users don't have Owner/Contributor permission. Do not terminate execution.
+                return true;
+            }
+
+            if (result.Port != Int32.Parse(Port))
+            {
+                return false;
+               
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Prompts the user if they would like to update the Service Configuration to allow the target port. If they confirm, perform the operation.
+        /// </summary>
+        /// <exception cref="AzPSApplicationException">User might not have permission to create service config. Fail in that case.</exception>
+        private void CreateServiceConfiguration()
+        {
+            System.Collections.ObjectModel.Collection<ChoiceDescription> choices = new System.Collections.ObjectModel.Collection<ChoiceDescription> { new ChoiceDescription("N"), new ChoiceDescription("Y") };
+            int userChoice = this.Host.UI.PromptForChoice($"The port {Port} that you are trying to connect to is not allowed for SSH connections for this resource. Would you like to update the current Service Confugration in the endpoint to allow connections to port {Port}?", "You must have owner or contributor roles in this resource to be able to change the service configuration. For more information see: http://aka.ms/ssharc/allow-ports", choices, 0);
```

Comment:
This would be easier to read using parameter list names, and multiple lines:

```csharp
int userChoice = Host.UI.PromptForChoice(
    caption: $"The port...",
    message: "You must have owner...",
    choices: choices,
    defaultChoices: 0);
```

Also PowerShell coding guidelines recommends not using the `this.` keyword when referencing class properties or methods.